src/core/app.lua

```diff
@@ -124,6 +128,88 @@ function configure (new_config)
    counter.add(configs)
 end
 
+-- Claims a name for a program so it can be identified by name by other processes.
+--
+-- The name given to the function must be unique, if a name has been used before by
+-- an active process the function will error displaying an appropriate error message.
+-- Any other process can enumerate programs by their name and PID using enumerate_named_programs.
+function claim_name(name)
+   function name_from_path(dir)
+      -- Extracts last file/folder name from path
+      -- e.g. "/var/run/snabb/hello" -> "hello"
+      local ds, de = string.find(dir, "/[^/]*$")
+      return string.sub(dir, ds+1, de)
+   end
+
+   configuration[name] = name
+   local namedir = "by-name/" .. name
+   local namedirqualified = named_program_root .. "/" .. name
+   local piddir = shm.root .. "/" .. S.getpid()
+   local backlinkdir = piddir.."/name-backref"
+
+   -- Verify that the by-name directory exists.
+   shm.mkdir(namedir)
+
+   -- Verify that we've not already claimed a name
```

Comment:
Let's keep a global variable instead of doing the readlink, e.g. instead do `assert(not claimed_name)`.  You can probably elide the error message too since this is a simple coding error.